scvi/external/cellassign/_model.py

```diff
@@ -0,0 +1,102 @@
+import logging
+
+import numpy as np
+import pandas as pd
+import torch
+
+from anndata import AnnData
+
+from scvi.data import register_tensor_from_anndata
+from scvi.dataloaders import AnnDataLoader
+from scvi.external.cellassign._module import CellAssignModule
+from scvi.lightning import TrainingPlan
+from scvi.model.base import BaseModelClass
+
+logger = logging.getLogger(__name__)
+
+
+class CellAssign(BaseModelClass):
+    """
+    Reimplementation of CellAssign for reference-based annotation [Zhang19]_.
+
+    Parameters
+    ----------
+    adata
+        single-cell AnnData object that has been registered via :func:`~scvi.data.setup_anndata`.
+    cell_type_markers
+        Binary marker gene matrix
+    use_gpu
+        Use the GPU or not.
+    **model_kwargs
+        Keyword args for :class:`~scvi.external.CellAssignModels`
+
+    Examples
+    --------
+    >>> # TODO
+    """
+
+    def __init__(
+        self,
+        adata: AnnData,
+        cell_type_markers: pd.DataFrame,
+        size_factor_key: str,
+        use_gpu: bool = True,
+        **model_kwargs,
+    ):
+        super().__init__(adata, use_gpu=use_gpu)
+        # check that genes are the same in cell_type_markers are present in the anndata
+        # anndata may have more
```

Comment:
I removed this code, but let's add something a bit different.

1. Check that adata and cell type markers contain the same genes
2. If they aren't in the same order, go ahead and sort the marker matrix inplace

after checking both contain same names,
`cell_type_markers = cell_type_markers.loc[adata.var_names]`